ompi/mca/mtl/ofi/mtl_ofi_types.h

```diff
@@ -64,75 +75,198 @@ typedef struct mca_mtl_ofi_component_t {
 
 /* send posting */
-#define MTL_OFI_SET_SEND_BITS(match_bits, contextid, source, tag, type) \
-    {                                                                   \
-        match_bits = contextid;                                         \
-        match_bits = (match_bits << 16);                                \
-        match_bits |= (uint64_t)source;                                 \
-        match_bits = (match_bits << 32);                                \
-        match_bits |= (MTL_OFI_TAG_MASK & tag) | type;                  \
+#define MTL_OFI_SET_SEND_BITS(match_bits, comm_id, source, tag)      \
+    {                                                                \
+        match_bits = comm_id;                                        \
+        match_bits = (match_bits << MTL_OFI_SOURCE_BIT_COUNT);       \
+        match_bits |= (uint64_t)(source & MTL_OFI_SOURCE_MASK);      \
+        match_bits = (match_bits << (MTL_OFI_TAG_BIT_COUNT           \
+                                     + MTL_OFI_PROTO_BIT_COUNT));    \
+        match_bits |= (tag & MTL_OFI_TAG_MASK);                      \
     }
 
 /* receive posting */
 /* Special tags are used for collective operations.
  * MPI_ANY_TAG should not match these special tags.
  * See ompi/mca/coll/base/coll_tags.h
  */
-#define MTL_OFI_SET_RECV_BITS(match_bits, mask_bits, contextid, source, tag) \
-    {                                                                   \
-        match_bits = 0;                                                 \
-        mask_bits  = MTL_OFI_PROTOCOL_MASK;                             \
-                                                                        \
-        match_bits = contextid;                                         \
-        match_bits = (match_bits << 16);                                \
-                                                                        \
-        if (MPI_ANY_SOURCE == source) {                                 \
-            match_bits = (match_bits << 32);                            \
-            mask_bits |= MTL_OFI_SOURCE_MASK;                           \
-        } else {                                                        \
-            match_bits |= (uint64_t)source;                             \
-            match_bits = (match_bits << 32);                            \
-        }                                                               \
-                                                                        \
-        if (MPI_ANY_TAG == tag) {                                       \
-            mask_bits |= 0x000000007FFFFFFFULL;                         \
-        } else {                                                        \
-            match_bits |= (MTL_OFI_TAG_MASK & tag);                     \
-        }                                                               \
+#define MTL_OFI_SET_RECV_BITS(match_bits, mask_bits, comm_id, source, tag) \
```

Comment:
This is quite long for a macro.  Is there a reason to not make this an inline function?  (some of the others above are also a little long)

Reply:
No really good reason, just extended the existing approach. OMPI prefers _inline_ for longer macros? sure, I can change it.

Reply:
I thought we were cautious about using _inline_ for long operations because of the jitter it can cause as compilers can't optimize the instruction cache as much as usual? @bwbarrett ?

Reply:
Macros vs. inline in this case is probably to-MAY-to vs. to-MAH-to: i.e., you're shoving a bunch of code inline, regardless of the mechanism.